src/array/utf8/mutable.rs

```diff
@@ -223,7 +223,145 @@ impl<O: Offset, P: AsRef<str>> FromIterator<Option<P>> for MutableUtf8Array<O> {
     }
 }
 
+// TODO Move to tests/
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::bitmap::Bitmap;
+
+    #[test]
+    fn test_extend_trusted_len_values() {
+        let mut array = MutableUtf8Array::<i32>::new();
+
+        array.extend_trusted_len_values(["hi", "there"].iter());
+        array.extend_trusted_len_values(["hello"].iter());
+
+        assert_eq!(array.values().as_slice(), b"hitherehello");
+        assert_eq!(array.offsets().as_slice(), &[0, 2, 7, 12]);
+        assert!(array.validity().is_none());
+    }
+
+    #[test]
+    fn test_extend_trusted_len() {
+        let mut array = MutableUtf8Array::<i32>::new();
+
+        // TODO Understand why the following is not possible
+        //array.extend_trusted_len([Some("hi"), Some("there")].into_iter());
+
+        array.extend_trusted_len(vec![Some("hi"), Some("there")].into_iter());
+        array.extend_trusted_len(vec![None, Some("hello")].into_iter());
+
+        let array: Utf8Array<i32> = array.into();
+
+        assert_eq!(array.values().as_slice(), b"hitherehello");
+        assert_eq!(array.offsets().as_slice(), &[0, 2, 7, 7, 12]);
+        assert_eq!(
+            array.validity(),
+            &Some(Bitmap::from_u8_slice(&[0b00001011], 4))
+        );
+    }
+}
+
 impl<O: Offset> MutableUtf8Array<O> {
+    /// Extends the [`MutableUtf8Array`] from an iterator of values of trusted len.
+    /// This differs from `extended_trusted_len` which accepts iterator of optional values.
+    #[inline]
+    pub fn extend_trusted_len_values<I, P>(&mut self, iterator: I)
+    where
+        P: AsRef<str>,
+        I: TrustedLen<Item = P>,
+    {
+        unsafe { self.extend_trusted_len_values_unchecked(iterator) }
+    }
+
+    /// Extends the [`MutableUtf8Array`] from an iterator of values of trusted len.
+    /// This differs from `extended_trusted_len_unchecked` which accepts iterator of optional
+    /// values.
+    /// # Safety
+    /// The iterator must be trusted len.
+    #[inline]
+    pub unsafe fn extend_trusted_len_values_unchecked<I, P>(&mut self, iterator: I)
+    where
+        P: AsRef<str>,
+        I: Iterator<Item = P>,
+    {
+        let (_, upper) = iterator.size_hint();
+        let additional = upper.expect("extend_trusted_len_values requires an upper limit");
+
+        // TODO How to avoid creating vec, how to assign capacity without iterating
+        // the entire input iterator
+        let mut values = vec![];
+        let mut offsets = Vec::with_capacity(additional);
+
+        let mut length = self.last_offset();
+
+        for item in iterator {
+            let bytes = item.as_ref().as_bytes();
+
+            length += O::from_usize(bytes.len()).unwrap();
+
+            values.extend_from_slice(bytes);
+            offsets.push(length);
+        }
```

Comment:
@jorgecarleitao Should I rewrite this function as a single loop?, to avoid creating `values` and `offset` vecs, this might mean we won't be utilizing MutableBuffer's `extend_from_trusted_len_iter_unchecked`